artemis-commons/src/main/java/org/apache/activemq/artemis/api/core/ActiveMQResourceLimitException.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.api.core;
+
+public final class ActiveMQResourceLimitException extends ActiveMQException {
+
```

Comment:
Probably should be an extension of the existing ActiveMQSessionCreationException so any old code with an explicit catch clause still gets caught

Reply:
Do you mean ActiveMQResourceLimitException extends ActiveMQSessionCreationException, and with RESOURCE_LIMIT_REACHED ActiveMQExceptionType? If so, older client still can't understand new exception by new ActiveMQExceptionType code when the code is passed back to client.

Reply:
Theyre independent comments. E.g. this is as well as the other

Reply:
> Probably should be an extension of the existing ActiveMQSessionCreationException so any old code with an explicit catch clause still gets caught

I don't find code with an explicit catch ActiveMQSessionCreationException clause except in the tests. Most of them are catching ActiveMQException(or other extended exception) and judge type code. And when queue limit is reached(not session), throwing an extension of ActiveMQSessionCreationException sounds not proper, wdyt?

Reply:
The point here isnt code within the broker but code in end user clients. People who are using core api directly.